lib/rust/shapely/macros/src/before_main.rs

```diff
@@ -21,8 +21,8 @@ fn mangle_name(name: &str) -> String {
 
 /// Functions exposed in WASM have to have unique names. This utility creates a name based on the
 /// location the function was defined at (module path, line number, column number).
-fn unique_name() -> String {
-    mangle_name(&root_call_path())
+fn unique_name(name: &impl std::fmt::Display) -> String {
```

Comment:
lets make it monomorphic. We do not need polymorphism here. Little compile time win, but still.